core/src/components/popover/utils.ts

```diff
@@ -0,0 +1,804 @@
+  }
+}
+
+/**
+ * Calculates where the arrow positioning
+ * should be relative to the popover content.
+ */
+const calculateArrowPosition = (
+  side: PositionSide,
+  arrowWidth: number,
+  arrowHeight: number,
+  top: number,
+  left: number,
+  contentWidth: number,
+  contentHeight: number,
+  isRTL: boolean
+) => {
+  /**
+   * Note: When side is left, right, start, or end, the arrow is
+   * been rotated using a `transform`, so to move the arrow up or down
+   * by its dimension, you need to use `arrowWidth`.
+   */
+  const leftPosition = { arrowTop: top + (contentHeight / 2) - (arrowWidth / 2), arrowLeft: left + contentWidth - (arrowWidth / 2) };
+
+  /**
+   * Move the arrow to the left by arrowWidth and then
+   * again by half of its width because we have rotated
+   * the arrow using a transform.
+   */
+  const rightPosition = { arrowTop: top + (contentHeight / 2) - (arrowWidth / 2), arrowLeft: left - (arrowWidth * 1.5) }
+
+  switch (side) {
+    case 'top':
+      return { arrowTop: top + contentHeight, arrowLeft: left + (contentWidth / 2) - (arrowWidth / 2) }
+    case 'bottom':
+      return { arrowTop: top - arrowHeight, arrowLeft: left + (contentWidth / 2) - (arrowWidth / 2) }
+    case 'left':
+      return leftPosition;
+    case 'right':
+      return rightPosition;
+    case 'start':
+      return (isRTL) ? rightPosition : leftPosition;
+    case 'end':
+      return (isRTL) ? leftPosition : rightPosition;
+    default:
+      return { arrowTop: 0, arrowLeft: 0 }
+  }
+}
+
+/**
+ * Calculates the required top/left
+ * values needed to position the popover
+ * content on the side specified in the
+ * `side` prop.
+ */
+const calculatePopoverSide = (
+  side: PositionSide,
+  triggerBoundingBox: ReferenceCoordinates,
+  contentWidth: number,
+  contentHeight: number,
+  arrowWidth: number,
+  arrowHeight: number,
+  isRTL: boolean
+) => {
+  switch (side) {
+    case 'top':
+      return {
+        top: triggerBoundingBox.top - contentHeight - arrowHeight,
+        left: triggerBoundingBox.left
+      }
+    case 'right':
+      return {
+        top: triggerBoundingBox.top,
+        left: triggerBoundingBox.left + triggerBoundingBox.width + arrowWidth
+      }
+    case 'bottom':
+      return {
+        top: triggerBoundingBox.top + triggerBoundingBox.height + arrowHeight,
+        left: triggerBoundingBox.left
+      }
+    case 'left':
+      return {
+        top: triggerBoundingBox.top,
+        left: triggerBoundingBox.left - contentWidth - arrowWidth
+      }
+    case 'start':
+      return {
+        top: triggerBoundingBox.top,
+        left: (isRTL) ? triggerBoundingBox.left + triggerBoundingBox.width + arrowWidth : triggerBoundingBox.left - contentWidth - arrowWidth
+      }
+    case 'end':
+      return {
+        top: triggerBoundingBox.top,
+        left: (isRTL) ? triggerBoundingBox.left - contentWidth - arrowWidth : triggerBoundingBox.left + triggerBoundingBox.width + arrowWidth
+      }
```

Comment:
```suggestion
    case 'start':
      return (isRTL) ? calculatePopoverSide('right', triggerBoundingBox, contentWidth, contentHeight, arrowWidth, arrowHeight, isRTL) : calculatePopoverSide('left', triggerBoundingBox, contentWidth, contentHeight, arrowWidth, arrowHeight, isRTL);
    case 'end':
      return (isRTL) ? calculatePopoverSide('left', triggerBoundingBox, contentWidth, contentHeight, arrowWidth, arrowHeight, isRTL) : calculatePopoverSide('right', triggerBoundingBox, contentWidth, contentHeight, arrowWidth, arrowHeight, isRTL);
```

A little recursion helps us stay DRY here.